yamux/Cargo.toml

```diff
@@ -16,6 +16,7 @@ nohash-hasher = "0.2"
 parking_lot = "0.12"
 rand = "0.8.3"
 static_assertions = "1"
+pin-project = "1.1.0"
```

Comment:
Instead of using `pin-project`, we could as well implement `Unpin` for `Stream`.

``` rust
impl Unpin for Stream {}
```

That said, I doubt there is any async Rust code-base without `pin-project` in its dependency tree already. Thus fine to include it here as well.